### DISCUSSION TITLE
e220 spreading factory issue

### DISCUSSION THREAD

**Discussion by frogg1ven** [jgromes/RadioLib#405] [General]
Hi, I'm working on project where I'm using code similar to SX126x_Receive_Interrupt.ino, but I wanted to use ***e220***-900m30s, so I declared class like that:
`LLCC68 radio = new Module(5, 25, 32, 33);`
In my previous code (where I used ***e22***-900m30s) I initialized lora like this:
```
int state = radio.begin();
radio.setFrequency(869.5);
radio.setBandwidth(125.0);
radio.setSpreadingFactor(9);
radio.setOutputPower(22);
```
and it works fine, so I tried here do it same way, but I have status code -707.
I found in some other post that this may work:
`int state = radio.begin(869.5, 125.0, 9, 7, SX126X_SYNC_WORD_PRIVATE, 22, 8, 0);`
and with this initialization code is working but I'm not receiving data from transmiter.
Transmiter (on sx1262, ***e22***) initialization code here:
```
int state = radio.begin();
radio.setFrequency(869.5);
radio.setBandwidth(125.0);
radio.setSpreadingFactor(12);
radio.setOutputPower(19);  
```
I would like to ask about two things, first what is the difference between one line and set functions initializations, 
and second, what may be the problem that my receiver is not racaive any data? With two e22 its working great.
Could you help me please? 
Regards.

**Comment by jgromes** [comment]
> what is the difference between one line and set functions initializations

The single line begin method sets TCXO voltage to 0 (among other things), which disables TCXO and enables XTAL. You have to select correct crystal type in order for the module to be able to perfrom frequency calibration, so your module is probably using XTAL.

> what may be the problem that my receiver is not racaive any data

You're transmitting at spreading factor 12, and trying to receive at spreading factor 9, which is not possible. All devices in the network must use the same spreading factor.

**Reply by frogg1ven** [reply]
thanks for respond @jgromes , 
I have tested some things and:

> The single line begin method sets TCXO voltage to 0 (among other things), which disables TCXO and enables XTAL

Can I change that in set function? Because `radio.setTCXO(0);` still causing status code -707.
> You're transmitting at spreading factor 12, and trying to receive at spreading factor 9, which is not possible. All devices in the network must use the same spreading factor.

Changing spreading factory from 12 to 9 in transmitter worked, and now it is receiving correctly.

I have also tried to set output power to 30 dBm, and it's not working with ERR_INVALID_OUTPUT_POWER   -13 status code. I know that in settings allowed values are in range from -17 to 22 dBm, but E220 has max output power equal to 30, is there any way to set it to higher values?

Thank you for your fast support,
Regards.

**Reply by jgromes** [reply]
> radio.setTCXO(0); still causing status code -707.

IIRC TCXO can't be disabled once enabled.

> I have also tried to set output power to 30 dBm

See https://github.com/jgromes/RadioLib/wiki/High-power-Radio-Modules-Guide

**Reply by frogg1ven** [reply]
thanks for help

**Reply by frogg1ven** [reply]
@jgromes 
So for high power radio devices I have to ` setOutputPower(27)` and tx enable to reach 27 dBm of output power right? How to set 27 dBm of output power on e220-900m30s? How to calculate it?

**Reply by jgromes** [reply]
No, you have to set output power based on the power amplifier your module is using. For example, E22 has a +26 dB amplifier, so to get +30 dBm of output power, you have to `setOutputPower(4);`, becuase 4 + 26 = 30.

You're using E220, and I don't know if it's using the same power amplifier as E22, but I imagine it might, since it's for the same freqency band.

**Reply by frogg1ven** [reply]
thanks for reply, I'll check that

**Reply by frogg1ven** [reply]
@jgromes 
E220-900m30s has yp2233w power amplifier, if you need to know
![IMG_20211118_171644](https://user-images.githubusercontent.com/49206712/142458066-37f3cf29-c230-4807-a18f-cc2d2f69b768.jpg)
Should I `setOutputPower(1)` and TX enable to achieve 27 dBm of output power, right?

**Reply by jgromes** [reply]
Thanks for the photo, it looks exactly like my E22 and has the same PA, so that confirms that theory.

> Should I setOutputPower(1) and TX enable to achieve 27 dBm of output power, right?

Yes, that should be correct.

**Reply by frogg1ven** [reply]
thanks
